invenio_base/urls/builders.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2025 CERN.
+# Copyright (C) 2025 Northwestern University.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""InvenioUrlsBuilder.
+
+This code is in support of providing `invenio_url_for`, the RDM supplement
+to Flask's `url_for` which allows any part of the application (in the broad
+sense of the term) to generate URLs even if those URLs are for views registered
+in another Flask application. Other developer niceties are included.
+"""
+
+from abc import ABC, abstractmethod
+
+from flask import Flask, current_app
+from importlib_metadata import entry_points as iter_entry_points
+from werkzeug.routing import BuildError, Map, Rule
+
+from .proxies import current_app_map_adapter, other_app_map_adapter
+
+
+class InvenioUrlsBuilder(ABC):
+    """Interface of class in charge of producing urls."""
+
+    @abstractmethod
+    def build(self, endpoint, values, method=None):
+        """Build current or other app url."""
+
+
+class NoOpInvenioUrlsBuilder(InvenioUrlsBuilder):
+    """Doesn't do anything."""
+
+    def build(self, endpoint, values, method=None):
+        """Instead of building returns empty string."""
+        return ""
+
+
+class InvenioAppsUrlsBuilder(InvenioUrlsBuilder):
+    """Builds URLs with some knowledge of Invenio (app-rdm)."""
+
+    def __init__(
+        self,
+        cfg_of_app_prefix,
+        cfg_of_other_app_prefix,
+        groups_of_other_app_entrypoints,
```

Comment:
See previous commentary (still applicable, just outdated because of name change)